### DISCUSSION TITLE
Option to display filenames in gallery.

### DISCUSSION THREAD

**Discussion by Meatfucker** [immich-app/immich#10824] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I have a large album full of videos which only show black images as thumbnails and due to this are indistinguishable from each other. They are well named however and being able to display this information would make them identifiable.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Try rerunning thumbnail generation for those assets, in the latest releases Immich should now pick a better frame to use as the thumbnail.

**Reply by Meatfucker** [reply]
I appreciate the tip and will give that shot. That said I would still find the ability to display filenames (or even other metadata) in the gallery view to be very helpful at times.

**Comment by HansLau** [comment]
Yes, would be a good option to be able to view filename, perhaps at the bottom corner of the image

**Comment by Aghead-B** [comment]
@Meatfucker did you find a way yo view the filename in the gallery view?

**Comment by rpm5099** [comment]
This seams to be an issue on a lot of open source platforms.  I cannot figure out why, this seems like basic functionality, lots of things dont have embedded metadata especially older files.

**Comment by Jack25apr** [comment]
As a client (Browser) i've ended up with JS injection and single info requests for each unique object:

> 
> const STORAGE_KEY = 'immich-asset-names'; // storage key
> const cache = new Map(JSON.parse(localStorage.getItem(STORAGE_KEY) || '[]')); 
> const pending = new Map(); 
> const queue = []; // simple queue (rate limit)
> let active = 0;
> const MAX_CONCURRENT = 3;
> 
> function saveCache() { localStorage.setItem(STORAGE_KEY, JSON.stringify([...cache])); }
> 
> function fetchName(id) {
>   if (cache.has(id)) { return Promise.resolve(cache.get(id)); }
>   if (pending.has(id)) { return pending.get(id); }
>   const promise = new Promise((resolve) => { queue.push({ id, resolve }); runQueue(); });
>   pending.set(id, promise);
>   return promise;
> }
> function runQueue() {
>   if (active >= MAX_CONCURRENT || queue.length === 0) return;
>   const { id, resolve } = queue.shift();
>   active++;
>   fetch(`/api/assets/${id}`)
>     .then(r => r.json())
>     .then(data => {
>       const name = data.originalFileName;
>       cache.set(id, name);
>       pending.delete(id);
>       saveCache();
>       resolve(name);
>     })
>     .catch(() => { pending.delete(id); resolve('error'); })
>     .finally(() => { active--; runQueue(); });
> }
> function processVisible() {
>   const items = document.querySelectorAll('[data-asset-id]');
>   items.forEach(item => {
>     const id = item.getAttribute('data-asset-id');
>     if (!id) return;
>     const container = item.querySelector('[data-thumbnail-focus-container]');
>     if (!container) return;
>     let overlay = container.querySelector('.my-overlay');
>     if (!overlay) {
>       container.style.position = 'relative';
>       overlay = document.createElement('div');
>       overlay.className = 'my-overlay';
>       container.appendChild(overlay);
>     }
>     if (overlay.dataset.assetId !== id) { 
>       overlay.dataset.assetId = id;
>       if (cache.has(id)) {
>         overlay.textContent = cache.get(id);
>       } else {
>         overlay.textContent = '...';
>         fetchName(id).then(name => {
>           if (overlay.dataset.assetId === id) { overlay.textContent = name; }
>         });
>       }
>     }
>   });
> }
> 
> const observer = new MutationObserver(() => { requestAnimationFrame(processVisible); }); 
> observer.observe(document.body, { childList: true, subtree: true });
> processVisible();
> const style = document.createElement('style');
> style.innerHTML = `.my-overlay{position:absolute;bottom:6px;left:6px;z-index:9999;color:white;font-size:16px;font-weight:bold;background:rgba(0,0,0,0.7);padding:3px 6px;border-radius:4px;pointer-events:none;}`;
> document.head.appendChild(style);

**Comment by valankar** [comment]
Another +1 for this. I was just trying to go through a set of pictures with someone who had all the image filenames asking to look at "IMG_9999.JPG" etc but was not possible to do so.

**Comment by mhdhamouday** [comment]
Any update here?

**Comment by joseant-jimenez** [comment]
It would be a great feature to have the option to display "description", filenames, date, etc, and see the album in some kind of grid view, with selected fields. I have an album of videos where I'm not able to identify each one. Thanks.
